### PR TITLE
Issue #3391286 - Update distribution to use the ckeditor5 

### PR DESCRIPTION
**Pull request by robertragas**
## Problem
Ckeditor4 is being deprecated, so we need to make sure to get ready for Ckeditor5

## Solution
- Update social_editor to use the new ckeditor5 from the drupal core
- Update the contrib modules to use ckeditor5
-- url_embed
-- embed
- Update the socialbase and socialblue theme to be compatible with ckeditor5

## Issue tracker
https://www.drupal.org/project/social/issues/3391286

## Theme issue tracker
*[Required if applicable] Paste a link to the drupal.org theme issue queue item, either from [socialbase](https://www.drupal.org/project/socialbase) or [socialblue](https://www.drupal.org/project/socialblue). If any other issue trackers were used, include links to those too.*

## How to test
- [ ] Install open_social from scratch
- [ ] After logging in as an admin see that the text-formats (admin/config/content/formats) are set to ckeditor5
- [ ] Login as a site-manager 
- [ ] Create new content using the ckeditor5 editor and see that nothing weird shows.
- [ ] Do the same as a normal user
- [ ] Enable social_embed
- [ ] Try again with different roles and embed content
- [ ] See that they all are embedded correctly
- [ ] Login as a administrator/sitemanager and go to admin/config/opensocial/embed-consent
- [ ] Change the setting to enforce consent
- [ ] Login again as a normal user and see you have to first give consent to see the video

- [ ] Try the same as above, but now first install Open Social on an older version.
- [ ] Checkout to the new version to test the upgrade path

### Before merge
- [ ] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

## Release notes
We now support ckeditor5. All the text formats coming from the distro should automatically switch to the ckeditor5 version.

## Change Record
Even though ckeditor4 should still work, we do not recommend switching back. This is because you manually need to re-add the social_embed icon as this not contain an automatic upgrade/downgrade path.

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by BiaInacio** [comment]
Hi @robertragas,
I would like to raise up that the margins on the content creation pages when the CKeditor 5 is enabled is kind odd, could you please check it?
image

**Comment by robertragas** [comment]
Note about the width issue, to fix this we will need to patch #108 from https://www.drupal.org/project/drupal/issues/3241295

Thing is that currently I can't add this patch to the PR as it will not apply to our Drupal 9.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

File: modules/social_features/social_embed/social_embed.install [hunk 1]
```diff
@@ -41,14 +39,55 @@ function social_embed_install(bool $is_syncing) {
...
+}
+
+/**
+ * Implements hook_uninstall().
+ *
+ * Remove embed button from the toolbar.
+ */
+function social_embed_uninstall() :void {
+  $config_factory = \Drupal::configFactory();
+  $configuration = $config_factory->getEditable('editor.editor.full_html');
```
Comment (vcsvinicius) [inline modules/social_features/social_embed/social_embed.install:73]: During the hook-install the embeb are be added to basic-html and full-html, but here only full-html is uninstalling.

**Review by vcsvinicius — CHANGES_REQUESTED** [review]
@robertragas I added a simple assignment.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by robertragas** [comment]
> @robertragas I added a simple assignment.

Added the change

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Review by vcsvinicius — APPROVED** [review]
Everything is fine to be merged

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by ronaldtebrake** [comment]
> Note about the width issue, to fix this we will need to patch #108 from https://www.drupal.org/project/drupal/issues/3241295
> 
> Thing is that currently I can't add this patch to the PR as it will not apply to our Drupal 9.

Should we maybe create a follow up for that linked to the D10 regression checks already? So we can fix it as part of regression checks when it's D10 installable? 🙈

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3541-nxxea3my7ijit6pgqyznic3jhlkwbwjz.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/6513f74d0687b34e4bd4dd41

**Comment by ronaldtebrake** [comment]
it broke main as patch apparantly didnt apply, so reverted.

**Pull request merged**